pyproject.toml

```diff
@@ -37,6 +37,8 @@ s3fs = ">=2021.11, <2024"
 fsspec = ">=2022.1"
 tinynetrc = "^1.3.1"
 multimethod = ">=1.8"
+kerchunk = ">=0.1.2"
+dask = {extras = ["complete"], version = ">=2023.5.0"}
```

Comment:
What do you think of making these extra dependencies? I don't use poetry very much, maybe looks like this?

```
[tool.poetry.extras]
distributed = ["kerchunk >=0.1.2", "dask[complete] >=2023.5.0"]
```

Reply:
Great point, I'd be happy to move these to optional dependencies, only use `dask`-specific code if `dask` is installed, etc. I was just adding them as hard dependencies to start as I wasn't sure what you and @betolink would prefer.

Reply:
This is awesome @jrbourbeau , I think what @MattF-NSIDC and you are describing makes sense, optional dependencies and only use Dask specific code if Dask is installed (if this is not too much of a hassle)

Reply:
FWIW my general recommendation in cases like these is to depend on core `dask` (it's [trivially lightweight](https://github.com/dask/dask/blob/680b4daa1c20cf0d1b4cfa78aa90b263d40f3449/pyproject.toml#L29-L45)) and then use `dask.delayed` and `dask.compute`.  This will pick up a dask.distributed cluster if one is available, and if not use a local thread pool.